compiler/util/traverse_ir.py

```diff
@@ -17,27 +17,98 @@
 import inspect
 
 from compiler.util import ir_pb2
+from compiler.util import simple_memoizer
+
+
+class _FunctionCaller:
```

Comment:
It would be a more invasive change, but you could just require that all action functions have a `**kwargs` parameter, and skip all the reflection.

Reply:
That's a good point, I did consider it but as you noted:
1. It's more invasive
2. I found the error messages if you forget to do that a bit harder to understand

Going from 100K calls to 60 or so seems a reasonable tradeoff for perf and dev productivity.

Reply:
Only 60 calls to `getfullargspec`, but still a nontrivial amount of work on each invocation to filter the argument lists.

Either way: it's a big improvement over the status quo.